src/sql/parts/query/execution/queryActions.ts

```diff
@@ -417,6 +418,55 @@ export class ListDatabasesAction extends QueryTaskbarAction {
 	}
 }
 
+/**
+ * Action class that parses the query string in the current SQL text document.
+ */
+export class ParseSyntaxAction extends QueryTaskbarAction {
+
+	public static EnabledClass = '';
+	public static ID = 'parseQueryAction';
+
+	constructor(
+		editor: QueryEditor,
+		@IConnectionManagementService connectionManagementService: IConnectionManagementService,
+		@IQueryManagementService private _queryManagementService: IQueryManagementService,
+		@INotificationService private _notificationService: INotificationService,
+	) {
+		super(connectionManagementService, editor, ParseSyntaxAction.ID, ParseSyntaxAction.EnabledClass);
+		this.enabled = true;
+		this.label = nls.localize('parseSyntaxLabel', 'Parse');
+	}
+
+	public run(): TPromise<void> {
+		if (!this.editor.isSelectionEmpty()) {
+			if (this.isConnected(this.editor)) {
+				let text = this.editor.getSelectionText();
+				if (text === '') {
+					text = this.editor.getAllText();
+				}
+				this._queryManagementService.parseSyntax(this.editor.connectedUri, text).then(result => {
+					if (result && result.parseable) {
+						this._notificationService.notify({
+							severity: Severity.Info,
+							message: nls.localize('queryActions.parseSyntaxSuccess', 'Command successfully parsed')
```

Comment:
What message does SSMS use?  Can we also use the same message?  Mainly, I think it should say something like `SQL script file parsed successfully`.  I would expect to see the SQL statements referred to as `Commands`

Reply:
SSMS says `Commands completed successfully.` Should we go that way or the one you suggested?

Reply:
Let's stick with the SSMS wording.  Thanks!